pkg/component/etcd/configmap/values_helper.go

```diff
@@ -0,0 +1,75 @@
+// Copyright (c) 2022 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package configmap
+
+import (
+	"fmt"
+	"strconv"
+	"strings"
+
+	druidv1alpha1 "github.com/gardener/etcd-druid/api/v1alpha1"
+	"github.com/gardener/etcd-druid/pkg/utils"
+	"k8s.io/utils/pointer"
+)
+
+// GenerateValues generates `configmap.Values` for the configmap component with the given parameters.
+func GenerateValues(etcd *druidv1alpha1.Etcd) *Values {
+	initialCluster := prepareMultiNodeCluster(etcd)
+	values := &Values{
+		EtcdName:                etcd.Name,
+		EtcdNameSpace:           etcd.Namespace,
+		EtcdUID:                 etcd.UID,
+		Metrics:                 etcd.Spec.Etcd.Metrics,
+		Quota:                   etcd.Spec.Etcd.Quota,
+		InitialCluster:          initialCluster,
+		ClientUrlTLS:            etcd.Spec.Etcd.ClientUrlTLS,
+		PeerUrlTLS:              etcd.Spec.Etcd.PeerUrlTLS,
+		ClientServiceName:       utils.GetClientServiceName(etcd),
+		ClientPort:              etcd.Spec.Etcd.ClientPort,
+		PeerServiceName:         utils.GetPeerServiceName(etcd),
+		ServerPort:              etcd.Spec.Etcd.ServerPort,
+		AutoCompactionMode:      etcd.Spec.Common.AutoCompactionMode,
+		AutoCompactionRetention: etcd.Spec.Common.AutoCompactionRetention,
+		ConfigMapName:           utils.GetConfigmapName(etcd),
+	}
+	return values
+}
+
+func prepareMultiNodeCluster(etcd *druidv1alpha1.Etcd) string {
```

Comment:
```suggestion
func prepareInitialCluster(etcd *druidv1alpha1.Etcd) string {
```

Suggesting a change as this will be used to create `initial-cluster` for a single node setup as well as a multi node setup